sidecar/proxy/monitor/registry.go

```diff
@@ -85,44 +92,70 @@ func (m *registry) Start() error {
 func (m *registry) poll() error {
 	// Get newest catalog from registry
 	latestCatalog, err := m.getLatestCatalog()
+
 	if err != nil {
 		logrus.WithError(err).Warn("Could not get latest catalog from registry")
 		return err
 	}
 
 	// Check for changes
-	if !m.catalogsEqual(m.cached, latestCatalog) {
+	if !m.catalogsEqual(m.hashed, latestCatalog) {
 		// Update cached copy of catalog
-		m.cached = latestCatalog
-
+		m.lock.Lock()
+		m.hashed = latestCatalog
+		m.lock.Unlock()
+		instances, _ := m.registryClient.ListInstances(client.InstanceFilter{})
```

Comment:
We should avoid querying the remote registry again in this case. We already have the list of instances acquired from `m.getLatestCatalog()`. We can have `m.getLatestCatalog()` return a list instead of a map, have another function to construct a map out of this list, and use the list for notifying the listeners.